spec/change_set_persisters/change_set_persister_spec.rb

```diff
@@ -1721,6 +1721,72 @@
       end
     end
 
+    context "when deleting a ScannedResource" do
+      before do
+        # Make preservation deletes not actually happen to simulate a versioned
+        # file store.
+        allow(Valkyrie::StorageAdapter.find(:google_cloud_storage)).to receive(:delete)
+        # This is a bug - right now all disk:// storage adapter IDs are going to
+        # this adapter, no matter what, so the above never gets called.
+        allow(Valkyrie::StorageAdapter.find(:disk)).to receive(:delete)
+      end
+
+      it "generates a tombstone" do
+        file = fixture_file_upload("files/example.tif", "image/tiff")
+        resource = FactoryBot.create_for_repository(:pending_scanned_resource, preservation_policy: "cloud", files: [file])
+        change_set = DynamicChangeSet.new(resource)
+        change_set.validate(state: "complete")
+
+        change_set_persister.save(change_set: change_set)
+
+        change_set = DynamicChangeSet.new(resource)
+        change_set_persister.delete(change_set: change_set)
+
+        tombstones = change_set_persister.query_service.find_all_of_model(model: Tombstone)
+        expect(tombstones).not_to be_empty
```

Comment:
Can we make some more assertions here to make sure there's just one tombstone and it points to the old resource?